src/bitcoin-cli.cpp

```diff
@@ -235,7 +236,13 @@ UniValue CallRPC(const std::string& strMethod, const UniValue& params)
     assert(output_buffer);
     evbuffer_add(output_buffer, strRequest.data(), strRequest.size());
 
-    int r = evhttp_make_request(evcon.get(), req.get(), EVHTTP_REQ_POST, "/");
+    // check if we should use a special wallet endpoint
+    std::string endpoint = "/";
+    if (GetArg("-wallet", "") != "") {
+        // always use v1 endpoint if -wallet has been provided
+        endpoint = "/v1/wallet/"+urlEncode(GetArg("-wallet", ""));
```

Comment:
`urlEncode` doesn't exist.  I think if you just squashed 6a99141 (the following commit) into this one, you'd be ok.